doc/aaa/TACACS+ Requirement.md

```diff
@@ -0,0 +1,161 @@
+	- Specifying command names containing slashes.
+	- Importing function definitions from the shell environment at startup.
+	- Parsing the value of SHELLOPTS from the shell environment at startup.
+	- Redirecting output using the ‘>’, ‘>|’, ‘<>’, ‘>&’, ‘&>’, and ‘>>’ redirection operators.
+	- Builtin commands:
+
+| **Command** | **Behavior**                |
+| -------- | -------------------------- |
+| . | Specifying a filename containing a slash as an argument.     |
+| history     | Specifying a filename containing a slash as an argument. |
+| hash     | Specifying a filename containing a slash as an argument to the -p option. |
+| exec     | Specifying a filename containing a slash as an argument to the -p option. |
+| deleting/adding     | Use the -f and -d options to the enable builtin. |
+| enable     | Using the 'enable' builtin command to enable disabled shell builtins. |
+| command     | Specifying the -p option to the 'command' builtin command. |
+| set     | Turning off restricted mode with ‘set +r’ or ‘set +o restricted’. |
+
+	- All these behavior disabled only for user input, command in script will not be affected, see here for more details: https://www.gnu.org/software/bash/manual/html_node/The-Restricted-Shell.html
+
+
+- Supported Authorization types:
+	- EXEC: user session authorization support. this happen when user login.
+	- Command: user run command in shell.
+
+- Failover:
+	- Authorization will happen before execute, if remote TACACS server not available, use local group based authorization as failover.
+
+
+## 1.3 Accounting
+ - Accounting when:
+ 	- User login to SONiC host.
+ 	- User run command on host: 
+		- Command start run.
+		- Command finish.
+		- For recursive command/script, only the top level command have Accounting.
+- Failover:
+	- Use syslog as backup when remote TACACS not avaliable.
+
+
+## 1.4 User script
+ - Any script in whitelist can run with Authorization and Accounting. 
+ - If user create a script, admin user can use config command add script to whitelist.
+
+## 1.5 Docker support
+ - Docker exec command will be covered by Authorization and Accounting. 
+ - SONiC AAA can't cover any command user run inside a docker.
+
+## 1.6 Multiple TACACS server
+ - Support config multiple TACACS server.
+ - When a server not available, will try next server as backup.
+ - When all server not available, use native failover solution.
+
+# 2 Configuration and Management Requirements
+## 2.1 SONiC CLI
+ - Enable/Disable TACACS Authorization/Accounting command
+```
+    config tacacs authorization enable
+    config tacacs authorization disable
+    config tacacs accounting enable
+    config tacacs accounting disable
+```
+
+ - Whitelist management command
+```
+    show config tacacs restrict command whitelist
+    Config tacacs restrict command whitelist remove <command>
+    Config tacacs restrict command whitelist add <command>
+```
+
+ - Counter command
+```
+    show tacacs counter
+    clear tacacs counter
```

Comment:
>clear [](http://example.com/codeflow?start=4&length=5)

`clear` is part of `ncurses-bin`. We'd better not to override.

Reply:
Fixed.

Reply:
We have a similar command `sonic-clear`. ref: [https://github.com/Azure/sonic-utilities/blob/888701b67fd4f1cc5b9da534a360048f93f263f4/setup.py#L157](https://github.com/Azure/sonic-utilities/blob/888701b67fd4f1cc5b9da534a360048f93f263f4/setup.py#L157)

Reply:
Fixed with sonic-clear command, will push update later.